src/domain/app/App.tsx

```diff
@@ -12,102 +9,71 @@ import NotEligible from '../registration/notEligible/NotEligible';
 import Profile from '../profile/Profile';
 import AccessibilityStatement from '../accessibilityStatement/AccessibilityStatement';
 import { userHasProfileSelector } from '../registration/state/RegistrationSelectors';
 import TermsOfService from '../termsOfService/TermsOfService';
+import { authenticateWithBackend } from '../auth/authenticate';
+import SessionPrompt from './sessionPrompt/SessionPrompt';
+import { isSessionExpiredPromptOpen } from './state/ui/UISelectors';
 
-type AppProps = RouteComponentProps<{ locale: string }> & {
-  isLoadingUser: boolean;
-  userHasProfile: boolean;
-  fetchApiToken: (accessToken: string) => void;
-  fetchApiTokenError: (errors: object) => void;
-};
-
-class App extends React.Component<AppProps> {
-  componentDidMount() {
-    loadUser(store, userManager)
-      .then(user => {
-        if (user) {
-          this.props.fetchApiToken(user.access_token || '');
-        } else {
-          this.props.fetchApiTokenError({ message: 'No user found' });
-        }
-      })
-      .catch(error => {
-        // TODO: Clear oidc local storage when this happens.
-        toast(i18n.t('authentication.loadUserError.message'), {
-          type: toast.TYPE.ERROR,
-        });
-        this.props.fetchApiTokenError(error);
-        Sentry.captureException(error);
-      });
-  }
-
-  public render() {
-    const {
-      isLoadingUser,
-      userHasProfile,
-      match: {
-        params: { locale },
-      },
-    } = this.props;
+const App: React.FunctionComponent = props => {
+  const isLoadingUser = useSelector(isLoadingUserSelector);
+  const { locale } = useParams<{ locale: string }>();
+  const userHasProfile = useSelector(userHasProfileSelector);
+  const isSessionPromptOpen = useSelector(isSessionExpiredPromptOpen);
+  const apiToken = useSelector(apiTokenSelector);
+  const dispatch = useDispatch();
+  const user = useSelector(userSelector);
 
-    return (
-      <LoadingSpinner isLoading={isLoadingUser}>
-        <Switch>
-          <Redirect exact path={`/${locale}/`} to={`/${locale}/home`} />
-          <Route exact path={`/${locale}/home`} component={Home} />
-          <Route
-            exact
-            path={`/${locale}/registration/not-eligible`}
-            component={NotEligible}
-          />
-          <Route
-            exact
-            path={`/${locale}/accessibility`}
-            component={AccessibilityStatement}
-          />
-          <Route exact path={`/${locale}/terms`} component={TermsOfService} />
-          {!userHasProfile && (
-            <PrivateRoute exact path={`/${locale}/registration/form`}>
-              <RegistrationForm />
-            </PrivateRoute>
-          )}
-          <PrivateRoute exact path={`/${locale}/registration/success`}>
-            <Welcome />
-          </PrivateRoute>
+  useEffect(() => {
+    if (!apiToken && user?.access_token) {
+      dispatch(authenticateWithBackend(user.access_token));
+    }
+  }, [apiToken, dispatch, user]);
```

Comment:
This is better than componentDidMount, which only do token exchange when there are no token. 
In future when silent-renew work, we can remove the `!apiToken`, it will auto get new token when user data change.

(useEffect subscribe for apiToken / user changes)

Reply:
What if you add a comment about that future option?

Reply:
Sure things